pkg/vault/contracts/VaultAdmin.sol

```diff
@@ -498,30 +497,27 @@ contract VaultAdmin is IVaultAdmin, VaultCommon, Authentication, VaultGuard {
         whenVaultBuffersAreNotPaused
         withInitializedBuffer(wrappedToken)
         nonReentrant
-        returns (uint256 issuedShares)
+        returns (uint256 amountUnderlyingRaw, uint256 amountWrappedRaw)
     {
         // Check wrapped token asset correctness.
         address underlyingToken = wrappedToken.asset();
         _ensureCorrectBufferAsset(wrappedToken, underlyingToken);
 
-        // Take debt for assets going into the buffer (wrapped and underlying).
-        _takeDebt(IERC20(underlyingToken), amountUnderlyingRaw);
-        _takeDebt(wrappedToken, amountWrappedRaw);
-
         bytes32 bufferBalances = _bufferTokenBalances[wrappedToken];
 
-        // The buffer invariant is the sum of buffer token balances converted to underlying. We use `previewRedeem` to
-        // convert wrapped to underlying, since `redeem` is an EXACT_IN operation that rounds down the result.
-        uint256 currentInvariant = bufferBalances.getBalanceRaw() +
-            wrappedToken.previewRedeem(bufferBalances.getBalanceDerived());
+        // To proportionally add liquidity to buffer, we need to calculate the buffer invariant ratio. It's calculated
+        // as the amount of buffer shares the sender wants to issue (which in practice is the value that the sender
+        // will add to the buffer, expressed in underlying token amounts), divided by the total shares of
+        // the buffer.
+        uint256 invariantRatio = exactSharesToIssue.divUp(_bufferTotalShares[wrappedToken]);
+        // Multiply the current buffer balance by the invariant ratio to calculate the amount of underlying and wrapped
+        // tokens to add, keeping the proportion of the buffer.
+        amountUnderlyingRaw = bufferBalances.getBalanceRaw().mulUp(invariantRatio);
+        amountWrappedRaw = bufferBalances.getBalanceDerived().mulUp(invariantRatio);
```

Comment:
```suggestion
        // Multiply the current buffer balance by the invariant ratio to calculate the amount of underlying and wrapped
        // tokens to add, keeping the proportion of the buffer.
        uint256 totalShares = _bufferTotalShares[wrappedToken];
        amountUnderlyingRaw = bufferBalances.getBalanceRaw().mulDivUp(exactSharesToIssue, totalShares);
        amountWrappedRaw = bufferBalances.getBalanceDerived().mulDivUp(exactSharesToIssue, totalShares);
```

This should mimic more closely the logic for add liquidity proportional. It's more accurate to multiply first, and since we're multiplying and dividing we don't need the usual fixed point scaling / unscaling by FP(1).

Reply:
Makes sense!